lib/puma/binder.rb

```diff
@@ -431,6 +423,10 @@ def close_listeners
       end
     end
 
+    def close_unix_paths
+      @unix_paths.each { |up| File.unlink(up) if File.exist? up }
```

Comment:
I think the original behavior didn't check for existence, correct?

Reply:
You mean the behavior that no one may be quite sure when it worked?

1. Puma unlinks some of the UNIXSocket files on startup.  Maybe there should be a test for that.

2. The time between a stop command being issued and this code running is indeterminate.

3. IMHO, raising an error in case a file doesn't exist that Puma no longer needs is something that shouldn't happen.